javalib/src/main/scala/java/util/concurrent/ConcurrentNavigableMap.scala

```diff
@@ -0,0 +1,125 @@
+package java.util.concurrent
+
+import java.util.NavigableMap;
+import java.util.NavigableSet;
+
+// Ported form jsr166, revision: 1.20
+
+/*
+ * Written by Doug Lea with assistance from members of JCP JSR-166
+ * Expert Group and released to the public domain, as explained at
+ * http://creativecommons.org/publicdomain/zero/1.0/
+ */
+
+/** A {@link ConcurrentMap} supporting {@link NavigableMap} operations, and
+ *  recursively so for its navigable sub-maps.
+ *
+ *  <p>This interface is a member of the <a href="{@docRoot
+ *  }/java.base/java/util/package-summary.html#CollectionsFramework"> Java
+ *  Collections Framework</a>.
+ *
+ *  @author
+ *    Doug Lea
+ *  @param <K>
+ *    the type of keys maintained by this map
+ *  @param <V>
+ *    the type of mapped values
+ *  @since 1.6
+ */
+trait ConcurrentNavigableMap[K, V]
+    extends ConcurrentMap[K, V]
+    with NavigableMap[K, V] {
+
+  /** @throws ClassCastException
+   *  @throws NullPointerException
+   *  @throws IllegalArgumentException
+   */
+  override def subMap(
+      fromKey: K,
+      fromInclusive: Boolean,
+      toKey: K,
+      toInclusive: Boolean
+  ): ConcurrentNavigableMap[K, V]
+
+  override def headMap(
+      toKey: K,
+      inclusive: Boolean
+  ): ConcurrentNavigableMap[K, V]
+
+  override def tailMap(
+      fromKey: K,
+      inclusive: Boolean
+  ): ConcurrentNavigableMap[K, V]
+
+  override def subMap(fromKey: K, toKey: K): ConcurrentNavigableMap[K, V]
+
+  override def headMap(toKey: K): ConcurrentNavigableMap[K, V]
+
+  override def tailMap(fromKey: K): ConcurrentNavigableMap[K, V]
+
+  /** Returns a reverse order view of the mappings contained in this map. The
+   *  descending map is backed by this map, so changes to the map are reflected
+   *  in the descending map, and vice-versa.
+   *
+   *  <p>The returned map has an ordering equivalent to {@link
+   *  java.util.Collections# reverseOrder ( Comparator )
```

Comment:
Scalafmt does not work well with JavaDoc comments - because the `{@link java.util.Collections#}` is spanned across multiple lines Scaladoc would fail to build it. In most of JSR166 stuff (e.g. ForkJoinPool) javadoc comments were completely removed because of multiple issues related to that. Alternative is to use ScalaDoc syntax using brackets, however in case of JavaLib we can just get rid of documentation. 
We still have official documentations, which would be typically used, while our generation of javadocs could be mostly used to compare which classess/methods are available in our implementation.

Reply:
thanks, I omitted document in 4d6911c.